python/ray/data/iterator.py

```diff
@@ -312,22 +322,21 @@ def iter_torch_batches(
             get_device,
         )
 
-        if collate_fn is not None and (dtypes is not None or device is not None):
+        if finalize_fn is not None and (dtypes is not None or device is not None):
             raise ValueError(
-                "collate_fn cannot be used with dtypes and device. It is expected that"
-                "the provided `collate_fn` will move the output Torch tensors to the"
+                "finalize_fn cannot be used with dtypes and device. It is expected that"
+                "the provided `finalize_fn` will move the output Torch tensors to the"
                 "appropriate dtype and device."
             )
 
-        if collate_fn is None:
-
+        if finalize_fn is None:
             # Automatically move torch tensors to the appropriate device.
             if device is None:
                 default_device = get_device()
                 if default_device.type != "cpu":
                     device = default_device
 
-            def collate_fn(batch: Union[np.ndarray, Dict[str, np.ndarray]]):
+            def finalize_fn(batch: Union[np.ndarray, Dict[str, np.ndarray]]):
                 return convert_ndarray_batch_to_torch_tensor_batch(
```

Comment:
This may not work. The output type of `collate_fn` is Any, so with the current API, the input type to `finalize_fn` may not necessarily be a numpy batch.

Reply:
I see, how about we change `finalize_fn` to take in `Any`, and we can do a type check to see if the input is of `Union[np.ndarray, Dict[str, np.ndarray]]`? In the case where the provided `collate_fn` does not return a `numpy` type, then they would also need to provide the corresponding `finalize_fn` to handle the `Any` input type.